sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -545,6 +538,66 @@ case class AdaptiveSparkPlanExec(
     this.inputPlan == obj.asInstanceOf[AdaptiveSparkPlanExec].inputPlan
   }
 
+  /**
+   * This method is a wrapper of `createQueryStagesInternal`, which deals with result stage creation
+   */
+  private def createQueryStages(
+    resultHandler: SparkPlan => Any,
+    plan: SparkPlan,
+    firstRun: Boolean): CreateStageResult = {
+    plan match {
+      case resultStage@ResultQueryStageExec(_, optimizedPlan, _) =>
```

Comment:
how can we hit this branch?

Reply:
when `withFinalPlanUpdate` is invoked multiple times, e.g., when we do `df.collect` multiple times.

Reply:
let's add a code comment to mention it.